website/server/models/group.js

```diff
@@ -985,6 +995,30 @@ schema.methods.removeTask = async function groupRemoveTask (task) {
   }, {multi: true}).exec();
 };
 
+// Returns true if the user has reached the spam message limit
+schema.methods.checkChatSpam = function groupCheckChatSpam (user) {
+  if (user.contributor && user.contributor.level >= SPAM_CONTRIBUTOR_LEVEL) {
+    return false;
+  }
+
+  let group = this;
```

Comment:
We don't need to assign this to a variable. We never go into a context where the reference to `this` changes and the logic is not so complex that it's confusing to do `this.chat` instead.

If you wanted to do:

```js
let chat = this.chat;
```

I would be fine with that.